monitoring/collector.go

```diff
@@ -0,0 +1,105 @@
+package monitoring
+
+import (
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+)
+
+type Collector struct {
+	kubeClient    *clientset.Clientset
+	lhClient      *lhclientset.Clientset
+	namespace     string
+	currentNodeID string
+
+	log *logrus.Logger
```

Comment:
Use logger as name to be consistent with the different controllers.
use logrus.FieldLogger as type, and pass in the base logger in the new NewCollector() call
see format of the different controllers as example.

Then you can add a method, if you need per metric logging with different fields for example.
log := getLoggerForMetrics(collector.logger, metric)

Reply:
Thank you! Fixed to use logging structure